File: java/src/main/java/com/crio/qcontest/services/QuestionService.java
package com.crio.qcontest.services;

import java.util.List;

import com.crio.qcontest.entities.Level;
import com.crio.qcontest.entities.Question;
import com.crio.qcontest.repositories.IQuestionRepository;

public class QuestionService implements IQuestionService {

    private final IQuestionRepository questionRepository;

    public QuestionService(IQuestionRepository questionRepository) {
        this.questionRepository = questionRepository;
    }

    @Override
    public Question createQuestion(String title, Level level, Integer difficultyScore) {
        Question question = new Question(title, level, difficultyScore);
        return questionRepository.save(question);
    }

    @Override
    public List<Question> getAllQuestionLevelWise(Level level) {
        if(level == null){
            return questionRepository.findAll();
        }
        return questionRepository.findAllQuestionLevelWise(level);
    }
}

File: python/src/services/QuestionService.py
from typing import List
from src.entities import Level, Question
from src.repositories import QuestionRepository

class QuestionService:
    def __init__(self, questionRepository: QuestionRepository) -> None:
        self._questionRepository = questionRepository

    def createQuestion(self, title: str, level: Level, difficultyScore: int) -> Question:
        q = Question(title,level,difficultyScore)
        return self._questionRepository.save(q)

    def getQuestions(self, level: Level) -> List[Question]:
        if level is None:
            return self._questionRepository.findAll()
        return self._questionRepository.findAllQuestionLevelWise(level)
